File: ruby/quicksort.rb
def quicksort(arry)
  return arry if arry.length <= 1
  pivot = arry.pop
  bigger = []
  smaller = []
  arry.each_with_index do |num, i|
    smaller.push(arry[i]) if arry[i] < pivot
    bigger.push(arry[i]) if arry[i] > pivot
  end
  return [quicksort(smaller), pivot, quicksort(bigger)].flatten
end

p quicksort([4,32,32,5,1])
p quicksort([5,3,21,4,100,1])

File: js/quicksort.js
// pick a pivot - choose the last one
// everything smaller in one arry
// everything larger in a different arry
// then call quicksort on each list recursively

function quicksort(arry){
  if(arry.length <= 1) return arry;
  var pivot = arry.pop();
  var bigger = []
  var smaller = []
  for(let i = 0; i < arry.length; i ++){
    if(arry[i] < pivot){
      smaller.push(arry[i])
    } else {
      bigger.push(arry[i])
    }
  }
  return [...quicksort(smaller), pivot, ...quicksort(bigger)]
 }

console.log(quicksort([2,3,15,3]))
